src/lib.rs

```diff
@@ -210,6 +211,36 @@ impl LogLevel for InfoLevel {
     }
 }
 
+/// Default to [`log::Level::Debug`]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct DebugLevel;
+
+impl LogLevel for DebugLevel {
+    fn default() -> Option<log::Level> {
+        Some(log::Level::Debug)
+    }
+}
+
+/// Default to [`log::Level::Trace`]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct TraceLevel;
+
+impl LogLevel for TraceLevel {
+    fn default() -> Option<log::Level> {
+        Some(log::Level::Trace)
+    }
+}
+
+/// Default to [`None`]
+#[derive(Copy, Clone, Debug, Default)]
+pub struct OffLevel;
```

Comment:
In #10, we talked about how to handle `--quiet` but neither the code nor the description addresses this.

Reply:
I added the suggestion of `hide`

> value_parser that is assigned to a function that conditionally returns [UnknownArgumentValueParser](https://docs.rs/clap/latest/clap/builder/struct.UnknownArgumentValueParser.html) when L::default().is_none()

For this one I looked around the docs, and am not very familiar with clap value parsers, or how to use them conditionally in conjunction with actions. I can keep investigating this, I just wanted to address the other items first, and maybe get some guidance here.

Reply:
This would be something like:
```rust
fn default_value_parser::<L: LogLevel>() -> clap::builder::ValueParser {
    if L::default().is_some() {
        clap::builder::UnknownArgumentValueParser::suggest("'--quiet' is the default").into()
    } else {
        clap::builder::value_parser!(u8).into()
    }
}
...
#[arg(value_parser = default_value_parser::<L>())]
```

Still not entirely sure its the right call to make *but* starting with an error is the safer choice as we can stop erroring without a breaking change but to start erroring is a breaking change.

Hmm, maybe instead we should make `LogLevel` in charge of this.  If we provide inherent impls, then its generally recognized as not a major breaking change.

This would instead look like
```rust
pub trait LogLevel {
    // ...

    fn hide_quiet() -> bool {
        Self::default().is_none();
    }
    fn hide_verbose() -> bool {
        false
    }
    fn quiet_value_parser() -> bool {
        if L::default().is_some() {
            clap::builder::UnknownArgumentValueParser::suggest("'--quiet' is the default").into()
        } else {
            clap::builder::value_parser!(u8).into()
        }
    }
    fn verbose_value_parser() -> bool {
        clap::builder::value_parser!(u8).into()
    }
} 
```